### DISCUSSION TITLE
Upgrading TailwindCSS in React

### DISCUSSION THREAD

**Discussion by realstoman** [tailwindlabs/tailwindcss#6622] [Help] [answered]
Hello  everyone,

Recently wanted to upgrade to TailwindCSS v3 in my React project, followed the the [ upgrade guide ](https://tailwindcss.com/docs/upgrade-guide)in the docs, but seems there are issues with the the PostCSS.
I get below error when running my server:

`./src/index.css (./node_modules/css-loader/dist/cjs.js??ref--5-oneOf-4-1!./node_modules/postcss-loader/src??postcss!./src/index.css)
Error: PostCSS plugin tailwindcss requires PostCSS 8.
Migration guide for end-users:
https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users`

I had configured the TailwindCSS v2 in my React project using **Craco** library

Is there a better way to upgrade to TailwindCSS v3?

@adamwathan @reinink @RobinMalfait

**Comment by NovusOpus** [comment]
Are you using the latest version of [CRA](https://github.com/facebook/create-react-app/releases/tag/v5.0.0)?

The version v5.0.0 have PostCSS 8(Finally!), and also support for [Tailwind CSS(Load the Tailwind PostCSS plugin if the project includes a tailwind.config.js file)](https://github.com/facebook/create-react-app/pull/11717).

**Reply by realstoman** [reply]
No, I'm using version 4 of CRA

**Reply by NovusOpus** [reply]
Update maybe? That's the easy way to solve it, since Tailwind CSS V3 only works with PostCSS 8, Unlike the v2 where there was a PostCSS 7 compatibility build.

**Reply by realstoman** [reply]
I did upgraded my CRA to v5, but still there were issues with PostCSS when running the sever

**Comment by NovusOpus** [comment, marked as answer]
If for some reason you cant upgrade the CRA to v5. Your other choice would be to use the [Tailwind CLI method](https://tailwindcss.com/docs/installation).

Here's a repo using the [Tailwind CLI Approach with CRA v4](https://github.com/adamwathan/cra-tailwind-3), shared by @adamwathan 

You can see all the changes he made on this [Commit](https://github.com/adamwathan/cra-tailwind-3/commit/e9d5b316ebfba08d74b90725e91cd3673ac5265e)

**Reply by realstoman** [reply]
Thank you @NovusOpus , this helped me to upgrade. Many thanks
